### PR TITLE
Add multi-compose

### PR DESCRIPTION
**Pull request by eshrubs**
This is similar to combine but works with async functions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,17 @@ CompletableFutures.combine(f1, f2, f3, f4, (a, b, c, d) -> a + b + c + d);
 CompletableFutures.combine(f1, f2, f3, f4, f5, (a, b, c, d, e) -> a + b + c + d + e);
 ```
 
+#### combine
```
Comment (mbruggmann) [inline README.md]: This should probably say "compose" (in the sentence below as well).

**Comment by mbruggmann** [comment]
@dflemstr @spkrka @pettermahlen Any strong opinions on this one? I think it's good!

File: src/main/java/com/spotify/futures/CompletableFutures.java [hunk 2]
```diff
@@ -360,6 +359,124 @@ public static void checkCompleted(CompletionStage<?> stage) {
...
   /**
+   * Composes multiple stages into another stage using a function.
+   *
+   * @param a        the first stage.
+   * @param b        the second stage.
+   * @param function the composing function.
+   * @param <R>      the type of the composed {@link CompletionStage}.
+   * @param <A>      the type of the first stage's value.
+   * @param <B>      the type of the second stage's value.
+   * @return a stage that is composed from the input stages using the function.
```
Comment (pettermahlen) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: Perhaps copy+modify the documentation from `CompletionStage.toCompletableFuture` stating:
```
@throw UnsupportedOperationException - if this implementation does not interoperate with CompletableFuture
```

Comment (pettermahlen) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: That would of course apply to all the methods.

**Comment by eshrubs** [comment]
@pettermahlen I agree with that sentiment. `compose` is like `combineAsync`. I kind of like sticking with the Java 8 `compose` terminology.

Do people have preferences for:

`compose`
`combineAsync`
`composeMany`

**Comment by pettermahlen** [comment]
@eshrubs Not sure I understand 

> I kind of like sticking with the Java 8 compose terminology.

I think that the Java 8 terminology for this is `combine`. `compose` composes/appends a function execution to a `CompletionStage`, whereas `combine` combines two stages. This PR introduces functionality that combines > 2 stages, so `combine` would be a better term, I think?

**Comment by eshrubs** [comment]
I can buy that!

I don't think `combine(...., NaryFunction<...., CompletableFuture<R>>)` would compile with the existing `combine(...., NaryFunction<...., R>)` so it will need a new name. 

Some suggestions then could be:
`combineFuture`
`composeMultiple`

**Comment by eshrubs** [comment]
@pettermahlen I have updated the patch. Please comment :)

@spkrka @mbruggmann Please review the new terminology

File: src/main/java/com/spotify/futures/CompletableFutures.java [hunk 3]
```diff
@@ -360,6 +359,132 @@ public static void checkCompleted(CompletionStage<?> stage) {
   }
 
   /**
+   * Composes multiple stages into another stage using a function.
+   *
+   * @param a        the first stage.
+   * @param b        the second stage.
+   * @param function the composing function.
```
Comment (pettermahlen) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: Change to 'combining' function

File: src/main/java/com/spotify/futures/CompletableFutures.java [hunk 4]
```diff
@@ -360,6 +359,132 @@ public static void checkCompleted(CompletionStage<?> stage) {
...
+        .thenCompose(ignored -> function.apply(af.join(), bf.join()));
+  }
+
+  /**
+   * Composes multiple stages into another stage using a function.
+   *
+   * @param a        the first stage.
+   * @param b        the second stage.
+   * @param c        the third stage.
+   * @param function the composing function.
```
Comment (pettermahlen) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: Change to 'combining' function

File: src/main/java/com/spotify/futures/CompletableFutures.java [hunk 5]
```diff
@@ -360,6 +359,132 @@ public static void checkCompleted(CompletionStage<?> stage) {
...
+  }
+
+  /**
+   * Composes multiple stages into another stage using a function.
+   *
+   * @param a        the first stage.
+   * @param b        the second stage.
+   * @param c        the third stage.
+   * @param d        the fourth stage.
+   * @param function the composing function.
```
Comment (pettermahlen) [inline src/main/java/com/spotify/futures/CompletableFutures.java]: Change to 'combining' function

**Comment by eshrubs** [comment]
Fixed!

**Pull request merged**